### ISSUE TITLE
Object not being mapped during Deep Clone

### ISSUE THREAD

**Issue by SteveWilkes** [eswann/Mapster#89]
Hi,

I've written [a project](https://github.com/agileobjects/AgileMapper/tree/master/AgileMapper.PerformanceTester) to test the performance of various mappers at different tasks, and it looks to have uncovered an issue in Mapster's deep-cloning.

The object being cloned has the following members (among others):

 - Foos : List<Foo>
 - FooArray : Foo[]
 - Ints : IEnumerable<int>
 - IntArray: int[]

As you can see from the screenshot, the post-mapping, cloned object has its FooArray elements and Ints properties copied, instead of mapped - that is, the values from the source object are being used directly instead of being cloned.

![2016-11-29 mapster cloning](https://cloud.githubusercontent.com/assets/8459425/20711078/ae965eba-b635-11e6-8350-49e320fa5f83.png)

Cheers,

Steve

**Comment by chaowlert** [comment]
for `foo.FooArray[0] == foo.Adapt<Foo>().FooArray[0]`, this cannot reproduce in current source. But it can be reproduced from latest package. I dig into history, this had been fixed since August, but it might not yet been published. (sorry for this)

For `IEnumerable<int>`, I intentional reuse the same object (for enumerable of primitive) just because I think that `IEnumerable` is readonly by type definition. Originally, I built Mapster for mapping poco to dto only. But this test case is valid, I will fix this. Thank you.
